FileIO/XmlIO/Boost/BoostXmlGmlInterface.cpp

```diff
@@ -58,222 +81,148 @@ bool BoostXmlGmlInterface::readFile(const std::string &fname)
+		readPoints(st, *points, *pnt_names);
+		_geo_objects.addPointVec(std::move(points), geo_name, pnt_names.release());
 	}
-	else
+
+	for (auto st : doc->getConfSubtreeList("polylines"))
 	{
-		delete ply_names;
+		readPolylines(st,
+		              *polylines,
+		              *_geo_objects.getPointVec(geo_name),
+		              _geo_objects.getPointVecObj(geo_name)->getIDMap(),
+		              *ply_names);
 	}
 
-	if (!surfaces->empty())
+	for (auto st : doc->getConfSubtreeList("surfaces"))
 	{
-		geo_objects->addSurfaceVec(std::move(surfaces), geo_name, sfc_names);
+		readSurfaces(st,
+		             *surfaces,
+		             *_geo_objects.getPointVec(geo_name),
+		             _geo_objects.getPointVecObj(geo_name)->getIDMap(),
+		             *sfc_names);
 	}
-	else
-	{
-		delete sfc_names;
+
+	if (!polylines->empty()) {
+		_geo_objects.addPolylineVec(std::move(polylines), geo_name, ply_names.release());
+	}
+
+	if (!surfaces->empty()) {
+		_geo_objects.addSurfaceVec(std::move(surfaces), geo_name, sfc_names.release());
 	}
 
 	return true;
 }
 
-void BoostXmlGmlInterface::readPoints(BaseLib::ConfigTree const& pointsRoot,
-	                                  std::vector<GeoLib::Point*>* points,
-	                                  std::map<std::string, std::size_t>* &pnt_names )
+void BoostXmlGmlInterface::readPoints(BaseLib::ConfigTreeNew const& pointsRoot,
+	                                  std::vector<GeoLib::Point*>& points,
+	                                  std::map<std::string, std::size_t>& pnt_names )
 {
-	BOOST_FOREACH( BaseLib::ConfigTree::value_type const & point, pointsRoot )
+	for (auto const pt : pointsRoot.getConfParamList("point"))
 	{
-		if (point.first.compare("point") != 0)
-			continue;
-
-		unsigned p_id = static_cast<unsigned>(point.second.get("<xmlattr>.id", std::numeric_limits<unsigned>::max()));
-		double   p_x  = static_cast<double>  (point.second.get("<xmlattr>.x",  std::numeric_limits<double>::max()));
-		double   p_y  = static_cast<double>  (point.second.get("<xmlattr>.y",  std::numeric_limits<double>::max()));
-		double   p_z  = static_cast<double>  (point.second.get("<xmlattr>.z",  std::numeric_limits<double>::max()));
-		std::string p_name = point.second.get("<xmlattr>.name", "");
-
-		if ( p_id == std::numeric_limits<unsigned>::max() || p_x == std::numeric_limits<double>::max() ||
-		     p_y  == std::numeric_limits<double>::max()   || p_z == std::numeric_limits<double>::max() )
-			WARN("BoostXmlGmlInterface::readPoints(): Skipping point, attribute missing in <point> tag:\n%s",
-				BaseLib::propertyTreeToString(point.second).c_str())
-		else
+		auto const p_id = pt.getConfAttribute<std::size_t>("id");
+		auto const p_x  = pt.getConfAttribute<double>("x");
+		auto const p_y  = pt.getConfAttribute<double>("y");
+		auto const p_z  = pt.getConfAttribute<double>("z");
+
+		auto const p_size = points.size();
+		insert_if_key_unique_else_error(_idx_map, p_id, p_size,
```

Comment:
I think the runtime behaviour of that call should be equal to the original solution, because there also an `_idx_map.insert()` wwas performed. If there is a huge number of points, then probably the overhead by `ConfigTreeNew` is much more severe. But huge numbers of points should certainly not be stored in plain XML files.